### PR TITLE
[enhancement] Can build statistics collection SQL across partitions and columns

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bugfix
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

    /*
     * Modified to split tasks at the partition and column levels,
     * and the number of rows to scan is the number of rows in the partition
     * where the column is located.
     * The number of rows is accumulated in turn until the maximum number of rows is accumulated.
     * Use UNION ALL connection between multiple tasks and collect them in one query
     */
## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by kangkaisen** [comment]
need comment

**Comment by Youngwb** [comment]
what the statistics collection SQL looks like now? could you give an example?

**Comment by HangyuanLiu** [comment]
> what the statistics collection SQL looks like now? could you give an example?

The format has not been modified in any way, only direct SQL across partitions can be connected using UNION ALL. An example can view the SQL at the code modification in the file AnalyzeStmtTest.java

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 31 / 39 (79.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|31|39|79.49%|[40, 41, 42, 43, 91, 92, 93, 94]|

**Review by Seaven — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java [hunk 1]
```diff
@@ -35,48 +37,69 @@ public List<Long> getPartitionIdList() {
 
     @Override
     public void collect(ConnectContext context) throws Exception {
+        for (String sql : buildCollectSQLList()) {
+            collectStatisticSync(sql, context);
+        }
+    }
+
+    /*
+     * Modified to split tasks at the partition and column levels,
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/statistic/FullStatisticsCollectJob.java:46]: why need the `Modified` word? This english comment isn't clear.

**Comment by kangkaisen** [comment]
What's the performance improvement for this PR?

**Comment by HangyuanLiu** [comment]
> What's the performance improvement for this PR?

tpcds_1t store_sales 6min
tpch_1t lineitem 7min

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 31 / 39 (79.49%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/statistic/FullStatisticsCollectJob.java|31|39|79.49%|[40, 41, 42, 43, 91, 92, 93, 94]|

**Review by Youngwb — APPROVED** [review]

**Pull request merged**